src/ServiceControl.Audit/Auditing/AuditPersister.cs

```diff
@@ -64,7 +65,7 @@ public async Task<IReadOnlyList<MessageContext>> Persist(List<MessageContext> co
                 var inserts = new List<Task>(contexts.Count);
                 foreach (var context in contexts)
                 {
-                    inserts.Add(ProcessMessage(context));
+                    inserts.Add(ProcessMessage(context, dispatcher));
```

Comment:
I looked at the code path and it seems it would be complex. The `IMessageSession` comes from the host that starts NSB but `IDispatchMessages` comes from the raw endpoint which lifecycle is managed by `AuditIngestion` class.

There is probably a lot of accidental complexity in `AuditImporter`, `AuditIngestion`, `AuditIngestionComponent`, `AuditIngestor` and `AuditPersister` classes but I think dealing with it would require a separate PR.